onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h

```diff
@@ -51,6 +52,7 @@ struct TensorrtFuncState {
   std::vector<std::vector<int>> input_info;
   std::vector<std::vector<int>> output_info;
   std::vector<std::vector<int64_t>> output_shapes;
+  OrtMutex* tensorrt_mu_ptr = nullptr;
```

Comment:
would it be better to use a reference instead of a pointer?

Reply:
I agree reference is safer. we may keep the pointer for now, later probably I need to change all the pointers in the structure to reference.